plugins/modules/idrac_reset.py

```diff
@@ -92,42 +167,253 @@
+            time.sleep(10)
+            retry_count = retry_count + 1
+            if retry_count == IDRAC_RESET_RETRIES:
+              self.module.exit_json(msg=LC_STATUS_MSG.format(lc_status= 'unreachable', retries=retry_count), unreachable=True)
+
+        if retry_count == IDRAC_RESET_RETRIES and lc_status_dict.get('LCStatus') != "Ready":
+            self.module.exit_json(msg=LC_STATUS_MSG.format(lc_status=lc_status_dict.get('LCStatus'), retries=retry_count),failed=True)
+    
+    def __create_output(self,status):
+      result = {}
+      tmp_res = {}
+      result['idracreset'] = {}
+      result['idracreset']['Data'] = {'StatusCode': status}
+      result['idracreset']['StatusCode'] = status
+      track_failed, wait_msg = None, None
+      self.staus_code_after_wait = 202
+      if status in STATUS_SUCCESS:
+        if self.wait_for_idrac:
+          track_failed, status_code, wait_msg = self.__wait_for_port_open()
+          self.staus_code_after_wait = status_code
+          if track_failed:
+            self.module.exit_json(msg=wait_msg, changed=True)
+        tmp_res['msg'] = IDRAC_RESET_SUCCESS_MSG
+        tmp_res['changed'] = True
+        result['idracreset']['Message'] = IDRAC_RESET_SUCCESS_MSG if self.wait_for_idrac else IDRAC_RESET_RESET_TRIGGER_MSG
+        result['idracreset']['Status'] = 'Success'
+        result['idracreset']['retVal'] = True
+      else:
+        tmp_res['msg']  = FAILED_RESET_MSG
+        tmp_res['changed'] = False
+        result['idracreset']['Message'] = FAILED_RESET_MSG
+        result['idracreset']['Status'] = 'FAILED'
+        result['idracreset']['retVal'] = False
+      return tmp_res, result
+
+    def __perform_operation(self, payload):
+      tmp_res, res = None, None
+      run_reset_status = self.idrac.invoke_request(IDRAC_RESET_RESET_URI, "POST", data=payload)
+      status = run_reset_status.status_code
+      tmp_res, res = self.__create_output(status)
+      return tmp_res, res 
+    
+    def __wait_for_port_open(self,interval=30):
+      timeout_wait = self.module.params.get('job_wait_timeout')
+      time.sleep(interval // 2)
+      msg = RESET_UNTRACK
+      wait = timeout_wait
+      track_failed = True
+      status_code = 503
+      while wait > 0:
+          try:
+              self.idrac.invoke_request(MANAGERS_URI, 'GET')
+              time.sleep(interval // 2)
+              msg = IDRAC_RESET_SUCCESS_MSG
+              track_failed = False
+              status_code= 200
+              break
+          except HTTPError as err:
+            status_code = err.code
+            if status_code == 401:
+              time.sleep(interval // 2)
+              msg = IDRAC_RESET_SUCCESS_MSG
+              track_failed = False
+              break
+          except URLError as err:
+              time.sleep(interval)
+              wait = wait - interval
+      return track_failed, status_code, msg
+             
+    def __reset_to_default_all(self):
+        payload = {}
```

Comment:
Remove all these functions and keep a single function which consume different parameters, as only reset_type is changing, reset all code is same. Will be easy to maintain in future.

Reply:
Done!